op-node/rollup/finality/finalizer.go

```diff
@@ -243,19 +243,23 @@ func (fi *Finalizer) tryFinalize() {
 	finalizedL2 := fi.lastFinalizedL2 // may be zeroed if nothing was finalized since startup.
 	var finalizedDerivedFrom eth.BlockID
 	// go through the latest inclusion data, and find the last L2 block that was derived from a finalized L1 block
+	fi.log.Debug("try finalize", "finality_data", fi.finalityData, "last_finalized_l2", finalizedL2)
 	for _, fd := range fi.finalityData {
 		if fd.L2Block.Number > finalizedL2.Number && fd.L1Block.Number <= fi.finalizedL1.Number {
 			lastFinalizedBlock := fi.findLastFinalizedL2BlockWithConsecutiveQuorom(fd.L2Block.Number, finalizedL2.Number)
 
 			// set finalized block(s)
 			if lastFinalizedBlock != nil {
-				finalizedL2, finalizedDerivedFrom = fi.updateFinalized(*lastFinalizedBlock, lastFinalizedBlock.L1Origin)
+				finalizedL2 = *lastFinalizedBlock
+				finalizedDerivedFrom = fd.L1Block
 			}
+			fi.log.Debug("set finalized block", "finalized_l2", finalizedL2, "finalized_derived_from", finalizedDerivedFrom, "fd_l2_block", fd.L2Block)
```

Comment:
if you see the original logic, this debug log is only called when finalizedL2 and finalizedDerivedFrom is set.

Reply:
yes, changed.

Reply:
no my point was "set finalized block" should be printed **only when** finalizedL2 and finalizedDerivedFrom is set

see the original code:
![image](https://github.com/user-attachments/assets/889b4ef0-e907-4f56-b5bf-9200cf6f5515)

but after your changes, it will be printed no matter whether they are changed or not